content/docs/1.4.4/advanced-resources/rwx-workloads.md

```diff
@@ -31,22 +31,55 @@ It is necessary to meet the following requirements in order to use RWX volumes.
 
+## Setting the volume mount options
+
+The volume must be mounted via NFS to be accessed, and by default Longhorn mounts with NFS version 4.1, with options `softerr`, a `timeo` value of 600, and a `retrans` value of 5.  If the NFS server becomes unreachable due to factors such as node power outages, network partitions and so on, NFS clients will fail an NFS request after the specified number of retransmissions, resulting in an NFS `ETIMEDOUT` error being returned to the calling application and potential data loss. If `softerr` is not supported, Longhorn will automatically revert to using the `soft` option instead, which will return an `EIO` in error conditions.
+
+For new volumes, users can override the defaults to use mount options of their choice by creating a customized StorageClass with an `nfsOptions` parameter. For instance:
+
+  ```yaml
+  kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: longhorn-test
+  provisioner: driver.longhorn.io
+  allowVolumeExpansion: true
+  reclaimPolicy: Delete
+  volumeBindingMode: Immediate
+  parameters:
+    numberOfReplicas: "3"
+    staleReplicaTimeout: "2880"
+    fromBackup: ""
+    fsType: "ext4"
+    nfsOptions: "vers=4.1,noresvport,softerr,timeo=600,retrans=5"
+  ```
+
+  Replace the `nfsOptions` string with a customized comma-separated list of legal options.  Then, create RWX volume PVCs using the above StorageClass.
+
+### Notes
+
+1. The complete set of options must be supplied.
+
+2. If there is a typo or an erroneous value in the string, Longhorn does no validation of the override and will not catch it.  The mount will be rejected by the NFS server, and the volume cannot be created or attached.
+
+3. In versions 1.4.0 to 1.4.3 and 1.5.0 to 1.5.1 of Longhorn, the Longhorn CSI plugin uses a `hard` mount option within the share-manager Pod in the `NodeStageVolume` step. The `hard` mount allows NFS requests to persistently retry without termination, ensuring that IOs do not fail. When the server is back online or a replacement server is recreated, the IOs resume seamlessly, thus guaranteeing data integrity. However, there is potential risk that to maintain file system stability, the Linux kernel will not permit unmounting a file system until all pending IOs are written back to storage. The system cannot undergo shutdown until all such file systems are unmounted. If the NFS server fails to recover successfully, the client nodes must undergo a forced reboot.
+
+    To mitigate the issue for existing volumes in those releases, upgrade to v1.4.4, v1.5.2, or anything later.  The `persistentvolume.spec.csi.volumeAttributes` field is immutable, and making adjustments to the `nfsOptions` within this field is prohibited.  After upgrading the Longhorn system, the default `softerr` or `soft` option will be automatically applied when reattaching RWX volumes, if there were no overrides.
```

Comment:
```suggestion
1. In Longhorn v1.4.0 to v1.4.3 and v1.5.0 to v1.5.1, volumes within a share manager pod (specifically, in `NodeStageVolume`) are hard mounted by the Longhorn CSI plugin. Hard mounting allows Longhorn to persistently retry sending NFS requests, ensuring that IOs do not fail even when the NFS server becomes inaccessible for some time. IOs resume seamlessly when the server regains connectivity or a replacement server is created.

  This mechanism for guaranteeing data integrity, however, comes with some risk. To maintain stability, the Linux kernel does not allow unmounting of a file system until all pending IOs are completed. This is a concern because the system cannot shut down until all file systems are unmounted. If the NFS server is unable to recover, the client nodes must undergo a forced reboot.

  To mitigate the issue, upgrade to v1.4.4, v1.5.2, or a later version. After upgrading, either `softerr` or `soft` is automatically applied to the `nfsOptions` parameter whenever RWX volumes are reattached (if the default settings are not overridden).
```